internal/mode/static/telemetry/collector.go

```diff
@@ -275,3 +263,56 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectNodeCount returns the number of nodes in the cluster.
+func CollectNodeCount(ctx context.Context, k8sClient client.Reader) (int, error) {
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return 0, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+
+	return len(nodes.Items), nil
+}
+
+type clusterInformation struct {
+	Platform  string
+	Version   string
+	ClusterID string
+}
+
+func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (clusterInformation, error) {
+	var clusterInfo clusterInformation
+	var err error
+
+	var nodes v1.NodeList
+	if err = k8sClient.List(ctx, &nodes); err != nil {
```

Comment:
@kate-osborn 
do we use inline error handling to avoid creating err variables in the surrounded code?

If that is case, let's use here and similar places:
```
if err := k8sClient.List(ctx, &nodes); err != nil {
```

and for this and similar cases, no need to do inline error handling:
```
	if clusterID, err = CollectClusterID(ctx, k8sClient); err != nil {
		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
	}
```
->
```
	clusterID, err := CollectClusterID(ctx, k8sClient)
        if err != nil {
		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
	}
```

Reply:
This makes sense 👍